Scenarios/Authentication/Mixed-auth-allow-anonymous-workaround/Service/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+
+var builder = WebApplication.CreateBuilder();
+
+builder.Services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+    .AddJwtBearer(JwtBearerDefaults.AuthenticationScheme, options =>
+    {
+        options.Authority = "https://localhost:7222/";
+        options.Audience = "api";
+    });
+builder.Services.AddAuthorization(options =>
+{
+    options.DefaultPolicy = new AuthorizationPolicyBuilder(JwtBearerDefaults.AuthenticationScheme)
+        .RequireAuthenticatedUser()
+        .RequireClaim("scope", "api")
```

Comment:
This will unfortunately break as soon as you'll grant a second scope: the standard representation of the scopes in the JWT access token specification is a unique `scope` claim formatted as a space-separated string.

To fix that, you'll probably want to either use some string splitting. Alternatively, if you use the OpenIddict validation handler, you can use `RequireAssertion(context => context.Principal.HasScope("api")`.

Reply:
Indeed. i wrote it like this as demo purpose but i've updated as one may copy/paste and have to figure out the issue.